packages/react-scripts/scripts/start.js

```diff
@@ -11,6 +11,11 @@
 
 process.env.NODE_ENV = 'development';
 
+// Load environment variables from .env file. Surpress warnings using silent
+// if this file is missing. dotenv will never modify any environment variables
+// that have already been set.
+require('dotenv').config({silent: true});
```

Comment:
I liked this idea, but I can't get it to work. I think somehow that `config/env.js` is not loaded or not loaded early enough:

```diff
diff --git a/packages/react-scripts/config/env.js b/packages/react-scripts/config/env.js
index 66acf11..08faa82 100644
--- a/packages/react-scripts/config/env.js
+++ b/packages/react-scripts/config/env.js
@@ -14,2 +14,8 @@

+// Load environment variables from .env file. Surpress warnings using silent
+// if this file is missing. dotenv will never modify any environment variables
+// that have already been set.
+// https://github.com/motdotla/dotenv
+require('dotenv').config({silent: true});
+
 var REACT_APP = /^REACT_APP_/i;
diff --git a/packages/react-scripts/config/webpack.config.dev.js b/packages/react-scripts/config/webpack.config.dev.js
index 8d1de1c..5596de6 100644
--- a/packages/react-scripts/config/webpack.config.dev.js
+++ b/packages/react-scripts/config/webpack.config.dev.js
@@ -11,2 +11,3 @@

+var env = require('./env');
 var path = require('path');
@@ -18,3 +19,2 @@ var WatchMissingNodeModulesPlugin = require('../scripts/utils/WatchMissingNodeMo
 var paths = require('./paths');
-var env = require('./env');

diff --git a/packages/react-scripts/config/webpack.config.prod.js b/packages/react-scripts/config/webpack.config.prod.js
index 5790e8e..5c793c8 100644
--- a/packages/react-scripts/config/webpack.config.prod.js
+++ b/packages/react-scripts/config/webpack.config.prod.js
@@ -11,2 +11,3 @@

+var env = require('./env');
 var path = require('path');
@@ -18,3 +19,2 @@ var url = require('url');
 var paths = require('./paths');
-var env = require('./env');

diff --git a/packages/react-scripts/scripts/build.js b/packages/react-scripts/scripts/build.js
index 7d87acc..71dcc79 100644
--- a/packages/react-scripts/scripts/build.js
+++ b/packages/react-scripts/scripts/build.js
@@ -14,8 +14,2 @@ process.env.NODE_ENV = 'production';

-// Load environment variables from .env file. Surpress warnings using silent
-// if this file is missing. dotenv will never modify any environment variables
-// that have already been set.
-// https://github.com/motdotla/dotenv
-require('dotenv').config({silent: true});
-
 var chalk = require('chalk');
diff --git a/packages/react-scripts/config/webpack.config.prod.js b/packages/react-scripts/config/webpack.config.prod.js
index 5790e8e..5c793c8 100644
--- a/packages/react-scripts/config/webpack.config.prod.js
+++ b/packages/react-scripts/config/webpack.config.prod.js
@@ -11,2 +11,3 @@

+var env = require('./env');
 var path = require('path');
@@ -18,3 +19,2 @@ var url = require('url');
 var paths = require('./paths');
-var env = require('./env');

diff --git a/packages/react-scripts/scripts/build.js b/packages/react-scripts/scripts/build.js
index 7d87acc..71dcc79 100644
--- a/packages/react-scripts/scripts/build.js
+++ b/packages/react-scripts/scripts/build.js
@@ -14,8 +14,2 @@ process.env.NODE_ENV = 'production';

-// Load environment variables from .env file. Surpress warnings using silent
-// if this file is missing. dotenv will never modify any environment variables
-// that have already been set.
-// https://github.com/motdotla/dotenv
-require('dotenv').config({silent: true});
-
 var chalk = require('chalk');
diff --git a/packages/react-scripts/scripts/start.js b/packages/react-scripts/scripts/start.js
index a82e68d..b0290bf 100644
--- a/packages/react-scripts/scripts/start.js
+++ b/packages/react-scripts/scripts/start.js
@@ -13,8 +13,2 @@ process.env.NODE_ENV = 'development';

-// Load environment variables from .env file. Surpress warnings using silent
-// if this file is missing. dotenv will never modify any environment variables
-// that have already been set.
-// https://github.com/motdotla/dotenv
-require('dotenv').config({silent: true});
-
 var path = require('path');
diff --git a/packages/react-scripts/scripts/test.js b/packages/react-scripts/scripts/test.js
index 0c123ec..d13aff3 100644
--- a/packages/react-scripts/scripts/test.js
+++ b/packages/react-scripts/scripts/test.js
@@ -11,8 +11,2 @@ process.env.NODE_ENV = 'test';

-// Load environment variables from .env file. Surpress warnings using silent
-// if this file is missing. dotenv will never modify any environment variables
-// that have already been set.
-// https://github.com/motdotla/dotenv
-require('dotenv').config({silent: true});
-
 const createJestConfig = require('./utils/createJestConfig');
```